### DISCUSSION TITLE
character added to primary key (bigint)

### DISCUSSION THREAD

**Discussion by mof4tw** [prisma/prisma#6626] [Q&A] [answered]
The issue I'm having is kind of difficult to google.

An 'n' is being appended to the primary key in the result.

Version: ^2.21.2
Database connector: Mysql
Database mariadb

Here's the troubling code.
```
let user = await db.users.findFirst({
  where: {
    id:1,
  },  
})
```
**Result:**
```
{
  id: 1n,
  name: 'John Doe',
  email: 'john@doe.com',
  email_verified_at: null,
  password: 'XX',
}
```
Where does that n in the id field come from?

I do have relations on that, as the model below shows, but it's the same problem if I doesn't include those.
The n is appended to id's in the relations as well.

However if I use a raw query, all good:

```
const email = 'john@doe.com'
const result = await db.$queryRaw`SELECT * FROM users WHERE email = ${email};`
```
**Result:**
```
{
    id: 1,
    name: 'John Doe',
    email: 'john@doe.com',
    email_verified_at: null,
    password: 'xx',
}
```

Here's my (generated) model:
```
model users {
  id                         BigInt         @id @default(autoincrement()) @db.UnsignedBigInt
  name                       String         @db.VarChar(255)
  email                      String         @unique @db.VarChar(255)
  email_verified_at          DateTime?      @db.Timestamp(0)
  password                   String         @db.VarChar(255)
  remember_token             String?        @db.VarChar(100)
  created_at                 DateTime?      @db.Timestamp(0)
  updated_at                 DateTime?      @db.Timestamp(0)
  team_id                    BigInt?        @db.UnsignedBigInt
  teams_teamsTousers_team_id teams?         @relation("teamsTousers_team_id", fields: [team_id], references: [id])
  teams_teams_user_idTousers teams[]        @relation("teams_user_idTousers")
  team_user                  team_user[]
  user_devices               user_devices[]

  @@index([team_id], name: "users_team_id_foreign")
}
```

What am I doing wrong ? Appreciate your assistance.

**Comment by janpio** [comment, marked as answer]
This is just how BigInts are "displayed" or represented in Javascript:
> A BigInt value, also sometimes just called a BigInt, is a bigint primitive, created by appending n to the end of an integer literal, or by calling the BigInt() constructor (but without the new operator) and giving it an integer value or string value.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt#description

So all good I would say.

**Reply by mof4tw** [reply]
Thanks for your reply, much appreciated.

Well I'm exposing the prisma results to (apollo) graphql, which complaints with the following error:
`Int cannot represent non-integer value: 1`

Considering that the rawquery returns the bigint just fine, is there a way to have prisma return it in a format without the n character and without sanitizing all the results with Number(). That would be a lof of unnecessary loops and hoops, which is road I rather not choose to go down.

I do realizing that perhaps this isn't an issue with Prisma, but with Graphql not supporting bigInt. Just looking looking for the best workaround. The other ORM I'm using doesn't have this issue eg. Sequelize (migrating from Sequelize to Prisma).

Cheers a lot.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
